includes/MediaWiki/Specials/SpecialWikibaseReconcileEdit.php

```diff
@@ -0,0 +1,149 @@
+<?php
+
+namespace MediaWiki\Extension\WikibaseReconcileEdit\MediaWiki\Specials;
+
+use HTMLForm;
+use MediaWiki\Extension\WikibaseReconcileEdit\MediaWiki\Request\EditRequestParser;
+use MediaWiki\Extension\WikibaseReconcileEdit\MediaWiki\Request\EditRequestSaver;
+use MediaWiki\Extension\WikibaseReconcileEdit\MediaWiki\WikibaseReconcileEditServices;
+use MediaWiki\Rest\LocalizedHttpException;
+use SpecialPage;
+use Status;
+use Wikibase\DataModel\Entity\PropertyId;
+use Wikibase\Lib\Store\EntityRevision;
+use Wikibase\Repo\WikibaseRepo;
+
+/**
+ * @license GPL-2.0-or-later
+ */
+class SpecialWikibaseReconcileEdit extends SpecialPage {
+
+	/** @var string */
+	private $reconcileDefaultJSON;
+
+	/** @var string */
+	private	$entityDefaultJSON;
+
+	/** @var EditRequestParser */
+	private $editRequestParser;
+
+	/** @var EditRequestSaver */
+	private $editRequestSaver;
+
+	public function __construct() {
+		parent::__construct( 'WikibaseReconcileEdit' );
+
+		$this->editRequestParser = WikibaseReconcileEditServices::getEditRequestParser();
+		$this->editRequestSaver = WikibaseReconcileEditServices::getEditRequestSaver();
+
+		$propertyId = new PropertyId( 'P1' );
+
+		$this->reconcileDefaultJSON = json_encode( [
+			EditRequestParser::VERSION_KEY => '0.0.1',
+			'urlReconcile' => $propertyId->serialize(),
+		], JSON_PRETTY_PRINT | JSON_UNESCAPED_SLASHES );
+
+		$this->entityDefaultJSON = json_encode( [
+			EditRequestParser::VERSION_KEY => '0.0.1/minimal',
+			'statements' => [
+				[
+					'property' => $propertyId->serialize(),
+					'value' => "http://example.com/",
+				],
+			],
+		], JSON_PRETTY_PRINT | JSON_UNESCAPED_SLASHES );
+	}
+
+	public static function factory(): self {
+		return new self();
+	}
+
+	public function getDescription(): string {
+		return $this->msg( 'wikibasereconcileedit-special-description' )->text();
+	}
+
+	protected function getGroupName(): string {
+		return 'wikibase';
+	}
+
+	public function execute( $subPage ) {
+		parent::execute( $subPage );
+		$this->requireLogin();
+		$this->showRequestForm();
+
+		$out = $this->getOutput();
+		$out->addModules( 'ext.WikibaseReconcileEdit' );
+	}
+
+	public function sendRequest( $data ) {
+		$data['reconcile'] = json_decode( $data['reconcile'], true );
+		$data['entity'] = json_decode( $data['entity'], true );
+
+		try {
+			$request = $this->editRequestParser->parseRequestBody( $data );
+			$status = $this->editRequestSaver->persistEdits(
+				[ $request ],
+				$data['token'],
+				$this->getContext()->getUser()
+			);
+		} catch ( LocalizedHttpException $exception ) {
+			return Status::newFatal( \Message::newFromKey( $exception->getMessageValue()->getKey() ) );
+		}
+
+		/** @var EntityRevision $entityRevision */
+		$entityRevision = $status->getValue()[0]['revision'];
+
+		$entityUrl = WikibaseRepo::getDefaultInstance()
+			->getEntityUrlLookup()
+			->getLinkUrl( $entityRevision->getEntity()->getId() );
```

Comment:
I suspect this should be `getFullUrl()`, not `getLinkUrl()` – looking at some [redirects in codesearch](https://codesearch.wmcloud.org/search/?q=%5C%24this-%3EgetOutput%5C(%5C)-%3Eredirect&i=nope&files=&excludeFiles=&repos=), most of the URLs seem to come from `getFullURL()`, and only a handful (none of them in MediaWiki core) from `getLinkURL()`.

Reply:
done